minitask/new_import_data.py

```diff
@@ -90,12 +51,39 @@ def search_index_test(elastic_search):
         }
     # output = elastic_search.search(index='news', doc_type='web_news', body=test1)
     print('\n', 'searching for keyword: ', q, " ", "in", " article ", position, '\n')
-    output = elastic_search.search(index='news', body=test1)
+    output = elastic_search.search(index='wiki', body=test1)
     print('searching finished with total time: ', output['took'], '\n')
     print('result: ', '\n')
     for hit in output['hits']['hits']:
         # print search result
         print ('match news title: ', hit['_source']['title'], '\n', 'match news link: ', hit['_source']['link'], '\n', '-------------------------------------------------------------')
 
+def locked_iter(it):
+    it = iter(it)
+    lock = threading.Lock()
+    while True:
+        try:
+            with lock:
+                value = next(it)
+        except StopIteration:
+            return
+        yield value
+
+def main():
+    global doc_id, lock
+    if not elastic_search.indices.exists(index='news'):
+        elastic_search.indices.create(index='news')
+    totalLines = 0
+    ssl._create_default_https_context = ssl._create_unverified_context
+    with urlopen("https://dumps.wikimedia.org/enwiki/20210820/enwiki-20210820-all-titles-in-ns0.gz") as r:
+        with gzip.GzipFile(fileobj=r) as f:
+            pool = Pool(10)
```

Comment:
**[Desired, Maintainability]** Probably a good idea to make the pool thread count a file constant and just reference it here
```python
# Put this at the top of the file somewhere
POOL_THREAD_COUNT = 10
# ... other stuff ...
pool = Pool(POOL_THREAD_COUNT )
```